tests/Unit/RequestPaginationTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Tests\Unit;
+
+use GuzzleHttp\Promise\PromiseInterface;
+use Illuminate\Support\LazyCollection;
+use Saloon\Contracts\Response;
+use Saloon\Tests\Fixtures\Connectors\PageRequestPaginatorConnector;
+use Saloon\Tests\Fixtures\Requests\PageGetSuperHeroesRequest;
+
+test('you can yield from a paginator', function (): void {
```

Comment:
Is this test specifically testing the LazyCollection? If so I would change your test title a little bit